stable/oauth2-proxy/values.yaml

```diff
@@ -131,6 +131,12 @@ readinessProbe:
   periodSeconds: 10
   successThreshold: 1
 
+# Configure Kubernetes security context for container
+# Ref: https://kubernetes.io/docs/tasks/configure-pod-container/security-context/
+securityContext:
+  enabled: false
+  runAsNonRoot: true
```

Comment:
Is there something needed in the oauth2_proxy `Dockerfile` for non-root support?

Reply:
I could not find anything in the oauth2 proxy that requires root permissions.
So we could enable the security context by default, if we want to?

I choose to make `runAsNonRoot` configurable instead of default and/or hardcoded, so that it would not break other peoples deployments. In case that they use an old version of the oauth2 proxy image, or a version that they have build themself.

Reply:
The oauth2_proxy dockerfile already sets a user for the application, see the [dockerfile](https://github.com/pusher/oauth2_proxy/blob/master/Dockerfile#L30).

This full fills the requirement from Kubernetes, that the the user should be a "number and not a "name".